packages/edit-site/src/components/sidebar-navigation-screen-pages/index.js

```diff
@@ -35,65 +41,85 @@ export default function SidebarNavigationScreenPages() {
 		'page'
 	);
 
-	const { setIsCreatePageModalOpened } = useDispatch( editSiteStore );
+	const [ showAddPage, setShowAddPage ] = useState( false );
+
+	const history = useHistory();
+	const { setPage } = unlock( useDispatch( editSiteStore ) );
+
+	const handleNewPage = ( { type, id } ) => {
+		setPage( {
+			context: { postType: type, postId: id },
+		} );
```

Comment:
Why do we need to both set page and push history, and I think the setPage can probably be removed.